qpid_bow/config.py

```diff
@@ -19,30 +21,50 @@ def configure(new_config: Mapping):
     config.update(new_config)
 
 
-def get_urls(urls: Optional[str] = None) -> List[str]:
-    """Retrieves server urls from one of the sources.
+def process_url(url: str) -> str:
+    """Processes a URL for usage with Qpid Proton.
 
-    The sources priority comes in the following order: passed arguments,
-    global config, AMQP_SERVERS environment variable.
+    - ActiveMQ amqp+ssl scheme is replaced with amqps.
+    - Adds username and password from config.
 
     Args:
-        urls: Comma-separated urls.
+        url: Input URL.
 
     Returns:
-        List[str]: Returns list of urls to connect to.
+        str: Processed URL.
     """
-    if urls:
-        return [url.strip() for url in urls.split(',')]
+    splited_url = urlsplit(url.strip())
+    if splited_url.scheme == 'amqp+ssl':
+        splited_url = splited_url._replace(scheme='amqps')
+
+    if ((not splited_url.username or not splited_url.password) and
+            'username' in config and 'password' in config):
+        user_pass = f"{config['username']}:{config['password']}@"
+        new_netloc = user_pass + splited_url.netloc
+        splited_url = splited_url._replace(netloc=new_netloc)
```

Comment:
Why `_replace`? 🤔

Reply:
It's a form of namedtuple, eventhough it's mangled with the _ this is officially documented as the way to create a new one with changes: https://docs.python.org/3/library/collections.html#collections.somenamedtuple._replace